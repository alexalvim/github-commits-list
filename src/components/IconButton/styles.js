import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

const Button = styled.button`
  background-color: ${Colors.mainColor};
  color: ${Colors.lightestColor};
  cursor: pointer;
  padding: ${Spaces.half} ${Spaces.base};
`;

export {
  Button
};