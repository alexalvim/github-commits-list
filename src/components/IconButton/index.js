import React from 'react';

import { Button } from './styles';

export default ({ Icon, onClick }) => (
  <Button onClick={onClick}>
    <Icon
      size={24}/>
  </Button>
)