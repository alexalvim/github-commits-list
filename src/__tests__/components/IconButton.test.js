import React from 'react';
import { shallow } from 'enzyme';
import { IoMdSearch } from 'react-icons/io'

import IconButton from '../../components/IconButton';

describe('Testing IconButton', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<IconButton Icon={IoMdSearch} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('call expected function', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<IconButton Icon={IoMdSearch} onClick={mockFunction} />);

    wrapper.simulate('click');
    expect(mockFunction).toHaveBeenCalled();    
  })
});