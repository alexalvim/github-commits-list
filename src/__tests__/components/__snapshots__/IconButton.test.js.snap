// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing IconButton should render correctly 1`] = `ShallowWrapper {}`;

exports[`Testing UserBox should render correctly 1`] = `ShallowWrapper {}`;
